### PR TITLE
CI Improvements - Publishing packages

### PR DESCRIPTION
**Pull request by PrimalPimmy**
# Pull request

## Description

- Added drop down menu feature on Draft-Release workflow which has options: major, minor, patch, rc
- Improved release process by no longer requiring to input old version and various other small changes
- All workspaces have the same version now
- Publish feature to crates.io (only tremor-common package for now)

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

### CODE REVIEW

File: .github/workflows/create-release.yaml [hunk 1]
```diff
@@ -47,4 +41,13 @@ jobs:
           prerelease: false
           body: ${{ steps.extract-release-notes.outputs.release_notes }}
         env:
-          GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}
+      - name: Publish packages to crates.io
```
Comment (Licenser) [inline .github/workflows/create-release.yaml:45]: I think the name here isn't quite correct

Comment (PrimalPimmy) [reply .github/workflows/create-release.yaml:45]: what should it be?

File: .github/workflows/draft-release.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+name: "Draft new release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      new-version:
+        type: choice
+        description: "Which version you'd like to release?"
+        options:
+        - major
```
Comment (Licenser) [inline .github/workflows/draft-release.yaml:10]: This is fine for `-rc.2` but the workflow a;sp needs to allow to go from version -> version+1-rc.1 -...> version+1-rc.N -> version+1

**Review by Licenser — APPROVED** [review]
🚀 nice!

File: .github/workflows/draft-release.yaml (hunk 2 above)
Comment (PrimalPimmy) [reply .github/workflows/draft-release.yaml:10]: I'll look into it in the next PR! thanks for pointing out!

**Pull request merged**